src/test/java/org/apache/sling/resourceresolver/impl/mapping/MapEntriesTest.java

```diff
@@ -370,26 +381,31 @@ public void test_vanity_path_updates() throws Exception {
     @Test
     public void test_vanity_path_registration_include_exclude() throws IOException {
         final String[] validPaths = {"/libs/somewhere", "/libs/a/b", "/foo/a", "/baa/a"};
-        final String[] invalidPaths = {"/libs/denied/a", "/libs/denied/b/c", "/nowhere"};
 
         final List<Resource> resources = new ArrayList<>();
         for(final String val : validPaths) {
             resources.add(getVanityPathResource(val));
         }
-        for(final String val : invalidPaths) {
-            resources.add(getVanityPathResource(val));
-        }
-
-
-        when(resourceResolver.findResources(anyString(), eq("sql"))).thenAnswer(new Answer<Iterator<Resource>>() {
-
-            @Override
-            public Iterator<Resource> answer(InvocationOnMock invocation) throws Throwable {
-                if (invocation.getArguments()[0].toString().contains("sling:vanityPath")) {
-                    return resources.iterator();
-                } else {
-                    return Collections.<Resource> emptySet().iterator();
-                }
+        when(resourceResolver.findResources(anyString(), eq("sql"))).thenAnswer((Answer<Iterator<Resource>>) invocation -> {
```

Comment:
Although you are checking that the query is what you expect, unless I missed something there's no test that validates the query syntax? I understand this is similar to the existing tests but I think it would be better to run that query against a content repository provided by Sling JCR Mocks and validate the results.
I don't know how much time you have for that - this way of testing probably works for now but IMHO running the query would create a test that's easier to maintain and understand. Maybe factor out the query generation to its own package-local helper class to avoid issues with private methods.

Reply:
As far as I know, the execution of querys isn't implemented anywhere in the testing code that we have a available. IIUC, we need an actual instance to be able to execute querys and thus validating the query syntax and also the proper execution of the query. 

What I've done is indeed making sure that the expected query is being executed. 

I think factoring out query code etc etc might be one mile too far for what we're trying to achieve in this ticket? For us, given our big version storage, it's quite important that those paths will not be considered anymore.

Reply:
If other reviewers agree to merge this with the tests as is I will not object. Maybe just add a comment to the test explaining that actually running the query would be better?

Reply:
Just added that comment anyway. :) 

@bdelacretaz  Do you know of a way to test querys with a real query engine, without booting a Sling instance?

Reply:
You could use SlingContext and back it for the test with a Oak Repo.

Reply:
I would be a bit cautious with pulling in too much into this test module. Bring in Sling Mocks in low-level modules has in my experience lead to various classpath clashes and overriding of depedencies.

I think validating the query is better done at the integration test level. Or maybe, if we want to be sure, we might want to try importing a query parser from Oak and checking that the query is valid. But I think that running the full query is too much at this level.

Reply:
The [jcr/repoinit/ExecutionOrderTest](https://github.com/apache/sling-org-apache-sling-jcr-repoinit/blob/master/src/test/java/org/apache/sling/jcr/repoinit/ExecutionOrderTest.java) is an example using SlingContext. I agree with Robert that such things might be better tested in integration tests, and as mentioned I won't object to committing the code without a query validation test.

Reply:
For validating queries we can probably wrap the logic from Oak's QueryEngineImpl#parseQuery ( https://github.com/apache/jackrabbit-oak/blob/fa54797c66e262d07e6e139526855ae8cb800bd1/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryEngineImpl.java#L144-L230 ), which basically delegates to the `SQL2Parser` and `XPathToSQL2Converter` classes. Those classes look pretty self-contained to me and we should be able to stub any dependencies they have - they don't seem used.

To me, this looks like a good idea for a separate module or bundle, so we can reuse it in other places where it may be desirable to do so.

Reply:
Indeed, it looks like `ResourceResolverType.JCR_OAK` is a solution that can actually parse and execute a query. Very cool! I've been using it in the project I'm working on and that's really nice. It's also nice that I was able to get com.day.cq.search working in the unit test, to test the predicate search that we're executing. But that's something outside Sling, of course.

I've created https://issues.apache.org/jira/browse/SLING-10742 to tackle this.